src/pharmpy/tools/amd/run.py

```diff
@@ -1118,6 +1134,51 @@ def validate_input(
         validate_allometric_variable(model, allometric_variable)
 
     # COVSEARCH
+    if mechanistic_covariates:
+        allowed_covariates = get_covariates_allowed_in_covariate_effect(model)
+        allowed_parameters = allowed_parameters = set(get_pk_parameters(model)).union(
+            str(statement.symbol) for statement in model.statements.before_odes
+        )
+        for c in mechanistic_covariates:
+            if isinstance(c, str):
+                if c not in allowed_covariates:
+                    raise ValueError(
+                        f'Invalid mechanistic covariate: {c}.'
+                        f' Must be in {sorted(allowed_covariates)}'
+                    )
+            else:
+                if len(c) != 2:
+                    raise ValueError(
+                        f'Invalid argument in `mechanistic_covariate`: {c}.'
+                        f' Tuples need to be given with one parameter and one covariate.'
+                    )
+                cov_found = False
+                par_found = False
+                for a in c:
+                    if a in allowed_covariates:
+                        if cov_found:
+                            raise ValueError(
+                                f'`mechanistic_covariates` contain invalid argument: got `{c}`,'
+                                f' which contain two covariates.'
+                                f' Tuples need to be given with one parameter and one covariate.'
+                            )
```

Comment:
Since this error is identical for all three if-statements, perhaps it can be stored as a variable.

Reply:
They are not identical but very similar since they are all about how the tuples are to be given.
They all start with `mechanistic_covariates contain invalid argument: got {c},` and end with the hint `Tuples need to be given with one parameter and one covariate.`

but the second line gives information on what is actually wrong.

- `which contain two covariates.`
- `' which contain two parameters.'`
- `' which contain {a}.'` (where a is the argument causing the error)

I can change it to store the first/last line as two separate variables if anything but that might be messy in its own way

Reply:
My bad, then this is better!